seatunnel-connectors-v2/connector-rocketmq/src/main/java/org/apache/seatunnel/connectors/seatunnel/rocketmq/source/RocketMqSourceSplitEnumerator.java

```diff
@@ -46,67 +43,76 @@
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Slf4j
 public class RocketMqSourceSplitEnumerator
-        implements SourceSplitEnumerator<RocketMqSourceSplit, RocketMqSourceState> {
+        implements SourceSplitEnumerator<RocketMQPartitionSplit, RocketMqSourceState> {
 
-    private static final long DEFAULT_DISCOVERY_INTERVAL_MILLIS = 60 * 1000;
-    private final Map<MessageQueue, RocketMqSourceSplit> assignedSplit;
     private final ConsumerMetadata metadata;
-    private final Context<RocketMqSourceSplit> context;
-    private final Map<MessageQueue, RocketMqSourceSplit> pendingSplit;
+    private final Context<RocketMQPartitionSplit> context;
     private ScheduledExecutorService executor;
     private ScheduledFuture scheduledFuture;
-    private final Object lock = new Object();
+    private final Map<MessageQueue, RocketMQPartitionSplit> assignedSplit;
+    private final Map<MessageQueue, RocketMQPartitionSplit> pendingSplit;
     // ms
     private long discoveryIntervalMillis;
+    Map<MessageQueue, Long> specificStartupOffsets;
+    private volatile boolean initialized;
+    private DefaultMQPullConsumer consumer;
+    private final Object lock = new Object();
+    /** The topic used for this RocketMQSource. */
+    private final List<String> topics;
+
+    private final long consumerOffsetTimestamp;
 
     public RocketMqSourceSplitEnumerator(
-            ConsumerMetadata metadata, SourceSplitEnumerator.Context<RocketMqSourceSplit> context) {
+            ConsumerMetadata metadata,
+            Context<RocketMQPartitionSplit> context,
+            long discoveryIntervalMillis) {
         this.metadata = metadata;
+        final RocketMqBaseConfiguration config = this.metadata.getBaseConfig();
+        this.topics = this.metadata.getTopics();
+        this.consumerOffsetTimestamp = this.metadata.getStartOffsetsTimestamp();
         this.context = context;
+        this.discoveryIntervalMillis = discoveryIntervalMillis;
         this.assignedSplit = new HashMap<>();
         this.pendingSplit = new HashMap<>();
+        specificStartupOffsets = this.metadata.getSpecificStartOffsets();
         // Set `rocketmq.client.logUseSlf4j` to `true` to avoid create many
         // `AsyncAppender-Dispatcher-Thread`
         System.setProperty("rocketmq.client.logUseSlf4j", "true");
+        initialRocketMQConsumer(config);
```

Comment:
Wouldn't it be better if we reused the implementation within RocketMqAdminUtil here?